src/utils/sources-tree.js

```diff
@@ -156,7 +156,10 @@ function isDirectory(url: Object) {
   const parts = url.path.split("/").filter(p => p !== "");
 
   // Assume that all urls point to files except when they end with '/'
-  return parts.length === 0 || url.path.slice(-1) === "/";
+  // Or directory node has children
+  return parts.length === 0 ||
+    url.path.slice(-1) === "/" ||
```

Comment:
Removing that condition breaks this test https://github.com/devtools-html/debugger.html/blob/master/src/utils/tests/sources-tree.js#L151

It is because `http://example.com/folder/c/` is no longer considered a folder.